db/migrate/20240724204821_add_unique_resource_index_to_roles.rb

```diff
@@ -0,0 +1,8 @@
+class AddUniqueResourceIndexToRoles < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+  verbose!
+
+  def change
+    add_index :roles, %i[resource_id resource_type], unique: true, algorithm: :concurrently
```

Comment:
This is another line of defense for attempts at e.g. updating an existing user's role via import.